Design/SANS/BeamCentreFinderDesign.md

```diff
@@ -0,0 +1,119 @@
+  handles degree for rotations. It also takes the reducer as an input. Note that the ReductionSingleton() is 
+  injected into the BeamCenterLogger.
+* CentreFinder is where the reduction takes place and where the residuals are calculated which are the quanitity
+  we want to minimize. The CentreFinder itself contains bare references to the ReductionSingleton() itself.
+
+### Things which can be reused with little effort
+* The CentrePositioner and the underlying factories need to be extended to detect when we want to use ANSTO.
+  Then we should switch to a XY-instance for the PositionProvider. This means we have to change the input slightly
+  to take something more general than a reducer and base our factory decision on this more general input. Once it
+  has decided for an ISIS instruement, we can make use of the current implementation.
+* The Beam Centre Logger should be injected and completly replaced by an Ansto implementation. We log into the ISIS
+  SANS GUI currently.
+
+### Things which need to be redesigned
+
+An abstract base class which can provide a more general interface could look like this
+```python
+class SANSBeamCentreFinderReductionWrapper(abc):
+   ...
+
+   @abstractmethod
+   def initialize(xstart, ystart)
+   '''
+   @returns beam coords, steps 
+   '''
+
+   @abstractmethod
+   def updateInternalCoordinates(coord1, coord2)
+   '''
+   An update step before to be called before we seek the centre
+   '''
+
+   @abstractmethod
+   def applyLimits(rlow, rupp)
+   '''
+   Applies the limits and needs to be done after a Seek step.
+   '''
+
+
+   @abstractmethod
+   def getSeekCentreReductionWrapper(rlow, rupp)
+   '''
+   @returns the wrapper for the appropriate reduction workflow, 
+            which itself is an abc
+   '''
+```
+
+``` python
+class SANSSeekCentreReductionWrapper(abc):
+   ...
+
+   @abstractmethod
+   def move(sign_policy, coord1, coord2)
+   '''
+   Moves the instrument in the workspace. ISIS will leave this as is
+   ANSTO needs to think what the right way of moving is
+   '''
+
+   @abstractmethod
+   def move(sign_policy, coord1, coord2)
+   '''
+   Moves the instrument in the workspace. ISIS will leave this as is
+   ANSTO needs to think what the right way of moving is.
+   '''
+
+
+   @abstractmethod
+   def getReducedSlices(workspace, ??can_workspace)
+   '''
+   This takes in a the workspace performs a reduction and provides four slices for UP/DOWN LEF/RIGHT
+   @returns 4 1D, sliced workspaces
+   '''
+   
+   @abstractmethod
+   def getReducedSlices(workspace, ??can_workspace)
+   '''
+   This takes in a the workspace performs a reduction and provides four slices for UP/DOWN LEF/RIGHT
+   @returns 4 1D, sliced workspaces
+   '''
+
+
+   @abstractmethod
+   def ceanUpSlice(workspace)
+   '''
+   Removes the EndNans from teh workspace
+   '''
+
+
+
+```
+
+
+### Things to clarify and potential difficulties
+*  
```

Comment:
A class diagram for the proposed solution would be useful. Either using the dot format. See [here](https://github.com/mantidproject/documents/blob/master/Design/DataProcessorAlgorithmUI/DataProcessingUserInterface.md#current-structure) a hand drawing photographed would be just fine too.